### ISSUE TITLE
Not every network is connected to the internet.  Bring Back the ISO!

### ISSUE THREAD

**Issue by sharkbit** [MicrosoftDocs/visualstudio-docs#1306]
I want the ability to download and install off line with out cert issues and crazy install scripts...  BRING BACK THE ISO.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 228076d1-94bf-c502-c018-08f7c0faf377
* Version Independent ID: c64a5cd5-6236-4192-2bfc-923aa20c94da
* Content: [Create an Offline Installation of Visual Studio - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/install/create-an-offline-installation-of-visual-studio)
* Content Source: [docs/install/create-an-offline-installation-of-visual-studio.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/install/create-an-offline-installation-of-visual-studio.md)
* Product: **visual-studio-dev15**
* GitHub Login: @TerryGLee
* Microsoft Alias: **tglee**

**Comment by nGAGEOnline** [comment]
The web-installer is seemingly taking forever to install VS2017 and freezes up my system for several seconds at a time. Please make .ISO available again.

**Comment by ddivito** [comment]
https://downloadly.ir/software/programming/microsoft-visual-studio-5/

**Comment by ddivito** [comment]
The [.](https://downloadly.ir/software/programming/microsoft-visual-studio-5/)ISO is not officially on the microsoft website

**Comment by TerryGLee** [comment]
@sharkbit, @nGAGEOnline, and @ddivito - Thanks for your feedback. We hear that you prefer an ISO, but we do not provide an ISO for the following reasons:

1. Visual Studio 2017 is a huge product that supports many types of development: from Android to UWP to C++ to Unity to Linux. If we were to still offer a 'full' install, it would come in at over 50GB. That's one huge ISO image!
1. We surveyed hundreds of people during the previews and RCs, and people told us that they had historically used an ISO for two reasons:
(i) because they wanted to download once and create a network install for enterprise deployment;
(ii) because they had a shaky internet connection and wanted to be sure they'd downloaded successfully before installing. A single monolithic ISO isn't the best solution for either of those scenarios - it's just the one that people are habituated to over all these years.
1. For the former (enterprise deployment), we have a full administrators' guide that provides detailed guidance on how to deploy the product: https://docs.microsoft.com/en-us/visualstudio/install/visual-studio-administrator-guide. This includes PowerShell scripts, examples of using the installer to create and update a network cache, descriptions of how to deploy Visual Studio in a fully offline environment, and so on.
1. For the latter, we've worked really hard to improve the robustness of the installer. The componentization work we've done means that the smallest install is one-tenth the size of Visual Studio 2015, making it far more likely to succeed. And other typical installs are smaller too. We download VS in small packages that are more likely to succeed, and we use multiple ways (WebClient, BITS, WinInet) to download those files to minimize problems with AV and proxy software. If you want to download first then install, you can cache the components locally and then install from there.
1. Because of the nature of our product (rapidly updating, lots of third-party components), an ISO is problematic. We don't have the rights to redistribute several large components (including Android), and others update on a rapid cadence that often includes critical security fixes, meaning an ISO image is incomplete, often outdated, and potentially not secure. 

@nGAGEOnline We're sorry to hear that you are experiencing issues with your system freezing up. Please consider using our [Report a Problem](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio-2017) tool so that we can gather repro steps and investigate this further.

**Comment by nGAGEOnline** [comment]
@TerryGLee I've actually managed to solve my issue. Unfortunately it's more of a work-around that not everybody is willing to do. But here it is: 

Recently, Microsoft has updated the content that is downloaded with the MediaCreationKit (or what it's called), for updating Windows or creating a bootable USB or ISO. The tool now (in my case) creates a bootable USB with a Windows 10 installer that includes all updates upto and including April 2018. 

I had already done 2 clean installs in the weeks prior (installs that required an additional 5hrs to get windows updates installed) but I still kept having issues and I'm pretty sure it's because one or more of the updates leading upto and/or including the April 2018 update.

It just happened that a friend required some help with a fresh install on his laptop last week and I noticed that the USB installation was slightly different from what I had seen just recently on my own machine. It made me realize that the installer had been updated to a newer version, which was obvious once we booted into Windows on his laptop. Turns out that a fresh install of windows without the tedious long update process, one which instantly gets you to April 2018, results in a stable running Windows 10 installation, whereas my old install that was incrementally updated, had major system-wide issues ever since the April 2018 update. At the moment, Windows is 100% stable and responsive again.

In summary: I suspect that others with issues may also have deeper issues with their Windows installation and it's actually not related to VS2017 installer. It's just a suspicion of course and probably can't be proven, but it's not inconceivable.
